### ISSUE TITLE
Improve PKGBUILD file for Arch Linux

### ISSUE THREAD

**Issue by gornostal** [Ulauncher/Ulauncher#685]
User [yochananmarqos](https://aur.archlinux.org/packages/ulauncher/#comment-779542) pointed out the following:

- A package already provides itself, no need for provides() and conflicts().
- libappindicator-gtk3 should be optional
- wmctrl is missing as an optional dependency, see [Hotkey In Wayland](https://github.com/Ulauncher/Ulauncher/wiki/Hotkey-In-Wayland).
- Checksums are missing for the tarball, do not skip them
- The service should be installed
- Line 34 & 35 are not necessary

Improved PKGBUILD and install file:

```bash
pkgname=ulauncher
pkgver=5.9.0
pkgrel=1
pkgdesc="Application launcher for Linux"
arch=('any')
url="https://ulauncher.io"
license=('GPL3')
depends=('libkeybinder3' 'gtk3' 'webkit2gtk' 'libnotify' 'gdk-pixbuf2' 'python-dbus'
         'python-levenshtein' 'python-pyinotify' 'python-websocket-client'
         'python-xdg' 'python-gobject')
makedepends=('python-distutils-extra')
checkdepends=('desktop-file-utils')
optdepends=('libappindicator-gtk3: tray icon'
            'wmctrl: fix window focus on Wayland')
install="$pkgname.install"
source=("https://github.com/Ulauncher/Ulauncher/releases/download/$pkgver/${pkgname}_${pkgver}.tar.gz"
        "https://github.com/Ulauncher/Ulauncher/raw/$pkgver/contrib/systemd/$pkgname.service")
sha256sums=('8d10ab9097235070dddf017e1e4c60d10696ed880887bcccfca6784c01fce3f5'
            'acf583f52933c0f01fb02a64f4103b235c1780d2203e644ed1c7033c22fde4a2')
 
prepare() {
    cd "$srcdir/$pkgname"
    find -iname "*.py" | xargs sed -i 's=\(^#! */usr/bin.*\)python3 *$=\1python='
}
 
build() {
    cd "$srcdir/$pkgname"
    python setup.py build
}
 
check() {
    cd "$srcdir/$pkgname"
    desktop-file-validate "build/share/applications/$pkgname.desktop"
}
 
package() {
    cd "$srcdir/$pkgname"
    
    # https://github.com/Ulauncher/Ulauncher/issues/521
    install -Dm644 "build/share/applications/$pkgname.desktop" -t \
        "$pkgdir/usr/share/applications"
        
    export PYTHONHASHSEED=0
    python setup.py install --root="$pkgdir" --optimize=1 --skip-build
    
    install -Dm644 "$srcdir/$pkgname.service" -t "$pkgdir/usr/lib/systemd/user"
}
```

```bash
post_install() {
     echo ""
     echo "Enable the service:"
     echo "systemctl --user enable --now ulauncher.service"
     echo ""
}
 
post_upgrade() {
     echo ""
     echo "Restart the service:"
     echo "systemctl --user daemon-reload"
     echo "systemctl --user restart ulauncher.service"
     echo ""
}
```

**Comment by friday** [comment]
I think "A package already provides itself, no need for provides() and conflicts()" assumes the package name is "ulauncher".

If you're going to use the same PKGBUILD for "ulauncher-git" I think you need them.

**Comment by friday** [comment]
I changed this now, so instead of maintaining a PKGBUILD template in the Ulauncher repos, we check out the git repo and overwrite just the specific values for the version and the download link. This way updates (like changing dependencies) can be done after a release, and we don't actually need any Arch container.

Everything else has been addressed separately except the checksums. I would rather do this after changing the build/release logic for Ulauncher 6.0, but I'll close this issue sincee I have another tracking issue for that.
